dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_fields.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React from 'react';
+import { EuiButtonIcon, EuiPanel, EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
+import { ConfigListEntry } from '../../../../../../../../common/types/explorer';
+import { VIS_CHART_TYPES } from '../../../../../../../../common/constants/shared';
+import {
+  AGGREGATIONS,
+  CUSTOM_LABEL,
+  GROUPBY,
+} from '../../../../../../../../common/constants/explorer';
+
+export const DataConfigPanelFields = ({
+  list,
+  sectionName,
+  visType,
+  addButtonText,
+  handleServiceAdd,
+  handleServiceRemove,
+  handleServiceEdit,
+}: any) => {
+  const isHeatMapAddButton = (name: string) => {
```

Comment:
Could we optimize this piece of code here to be something like below? 
```
if (!list || !isArray(list) || visType !== VIS_CHART_TYPES.HeatMap) return false;
return name === AGGREGATIONS ? list.length === 1 : list.length >= 2;
```

Reply:
Done